Experimental myriad instructions. (#92)
UCL/TDMS: doc/developers.md

spdlog::debug("Send help");
```

<details>
### Compiling on UCL's `myriad` nodes

**Warning**
These instructions are a bit experimental. Please use with care (and report anything that's wrong here)!

> If you want to test changes on UCL's [myriad](https://www.rc.ucl.ac.uk/docs/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions...

Author: Right. If it's not an easily solvable ticket, then I agree. I was hoping it might be fixed soon (hence not writing about the workaround).

> If you want to test changes on UCL's [myriad](https://www.rc.ucl.ac.uk/docs/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions...

Reviewer: 👻  these instructions work just fine for me on Myriad – no https woes

> If you want to test changes on UCL's [myriad](https://www.rc.ucl.ac.uk/docs/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions...

Reviewer 2: Maybe because you have spdlog already?

> If you want to test changes on UCL's [myriad](https://www.rc.ucl.ac.uk/docs/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions...

Reviewer: seemingly not.. 
[build_log.txt](https://github.com/UCL/TDMS/files/9559750/build_log.txt)

> If you want to test changes on UCL's [myriad](https://www.rc.ucl.ac.uk/docs/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions...

Author: - [ ] Wait for a reply from the RC support people.

> If you want to test changes on UCL's [myriad](https://www.rc.ucl.ac.uk/docs/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions...

Author: [RemedyForce N:05509640](https://ucl.my.salesforce.com/apex/bmcservicedesk__ssredirect?inc=a1N07000004GfkEEAS)

> If you want to test changes on UCL's [myriad](https://www.rc.ucl.ac.uk/docs/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions...

Author: I added a commit proposal. Not sure disabling git SSL verification is the best solution, but it's what RC support suggested.

```{.sh}
module purge
module load beta-modules
module load gcc-libs/9.2.0 compilers/gnu/9.2.0 xorg-utils matlab/full/r2021a/9.10 fftw/3.3.6-pl2/gnu-4.9.2 cmake/3.21.1
cd tdms
mkdir build; cd build

> cmake .. 
> ```

Reviewer 3: ```suggestion
cmake .. 
make install
```
```

In case we want to be super thorough, not important though :sweat_smile:

spdlog::debug("Send help");
```

> ### Compiling on UCL's myriad cluster

Reviewer 2: Note that this should apply to all UCL RC clusters, not just Myriad, although since this program uses at most OpenMP other systems would be an overkill (for example Kathleen is for multi-node jobs only).

Myriad is usually spelled with capital letter in UCL RC docs
```suggestion
### Compiling on UCL's Myriad cluster
```

<details>

  > **Warning**
  > These instructions are a bit experimental. Please use with care (and report anything that's wrong here)!

  

>   If you want to test changes on UCL's [myriad](https://www.rc.ucl.ac.uk/docs/Clusters/Myriad/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions...

Reviewer 2: ```suggestion
  If you want to test changes on UCL's [Myriad](https://www.rc.ucl.ac.uk/docs/Clusters/Myriad/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions
```

> 
>   
>   If you want to test changes on UCL's [Myriad](https://www.rc.ucl.ac.uk/docs/Clusters/Myriad/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions
>   ```{.sh}

Author: Sorry for the drip drip changes.

I had another thought, should we also mention ssh key forwarding?

```suggestion
  
  If you want to test changes on UCL's [Myriad](https://www.rc.ucl.ac.uk/docs/Clusters/Myriad/) (and/or don't have MATLAB on your pesonal machine) you can try these instructions.
  Firstly, you will probably want to [forward you ssh agent](https://stackoverflow.com/questions/12257968/) for your github ssh key.
  To do this, you first need to run the following your **_local_** machine:
  ```{.sh}
  ssh-add -L # check your ssh agent is running
  ssh-add /path/to/your/github/key/id_rsa
  ssh -o ForwardAgent=yes your_user@myriad.rc.ucl.ac.uk
  ```

  And once you're on Myriad:

  ```{.sh}
  git clone git@github.com:UCL/TDMS.git
```

Author: Or is that excessive?

  module purge
  module load beta-modules
  module load gcc-libs/9.2.0 compilers/gnu/9.2.0 xorg-utils matlab/full/r2021a/9.10 fftw/3.3.6-pl2/gnu-4.9.2 cmake/3.21.1

>   cd tdms

Author: ```suggestion
  cd TDMS/tdms
```

  mkdir build; cd build
  cmake .. 
  make install

>   ```

Reviewer 2: I can't say I like this solution.  Also, may educated guess is that it's setting `LD_LIBRARY_PATH` in the Matlab module which screws everything up.  Just look at the output of `for file in $(find $(dirname $(dirname $(which git)))/libexec/git-core -type f); do ldd ${file} | grep -i matlab; done` after loading the Matlab module.

>   ```

Author: Me neither, really. My only idea:
* Do you know if we can we make CMakeLists try a git pull over ssh w/ preference, and fallback to https if that fails?
